File: src/main/java/com/yerti/core/particles/ParticleManager.java
package com.yerti.core.particles;

import net.minecraft.server.v1_8_R3.EnumParticle;
import net.minecraft.server.v1_8_R3.PacketPlayOutWorldParticles;
import org.apache.commons.lang.Validate;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer;
import org.bukkit.entity.Player;
import org.bukkit.util.Vector;


public class ParticleManager {

    public static void sendColorParticle(Location location, int r, int g, int b) {
        PacketPlayOutWorldParticles particles = new PacketPlayOutWorldParticles(EnumParticle.FLAME, true, (float) location.getX(), (float) location.getY(), (float) location.getZ(), r, g, b, (float) 255, 0, 10);

        for (Player player : Bukkit.getOnlinePlayers()) {
            ((CraftPlayer) player).getHandle().playerConnection.sendPacket(particles);
        }

    }

    public static void drawLine(Location start, Location end, int r, int b, int g) {
        World world = start.getWorld();

        Validate.isTrue(world.equals(end.getWorld()), "Locations cannot be in different worlds.");

        double distance = start.distance(end);

        Vector v1 = start.toVector();
        Vector v2 = end.toVector();

        Vector vector = v2.clone().subtract(v1).normalize().multiply(2);

        int covered = 0;

        for (; covered < distance; v1.add(vector)) {
            PacketPlayOutWorldParticles particles = new PacketPlayOutWorldParticles(EnumParticle.FLAME, true, (float) v1.getX(), (float) v1.getY(), (float) v1.getZ(), r, g, b, (float) 255, 0, 10);

            for (Player player : Bukkit.getOnlinePlayers()) {
                ((CraftPlayer) player).getHandle().playerConnection.sendPacket(particles);
            }

            covered += 2;


        }
    }


}

File: src/main/java/com/yerti/core/particles/ParticleManager.kt
package com.yerti.core.particles

import net.minecraft.server.v1_8_R3.EnumParticle
import net.minecraft.server.v1_8_R3.PacketPlayOutWorldParticles
import org.apache.commons.lang.Validate
import org.bukkit.Bukkit
import org.bukkit.Location
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer

object ParticleManager {
    fun sendColorParticle(location: Location, r: Int, g: Int, b: Int) {
        val particles = PacketPlayOutWorldParticles(EnumParticle.FLAME, true, location.x.toFloat(), location.y.toFloat(), location.z.toFloat(), r.toFloat(), g.toFloat(), b.toFloat(), 255.toFloat(), 0, 10)
        for (player in Bukkit.getOnlinePlayers()) {
            (player as CraftPlayer).handle.playerConnection.sendPacket(particles)
        }
    }

    fun drawLine(start: Location, end: Location, r: Int, b: Int, g: Int) {
        val world = start.world
        Validate.isTrue(world == end.world, "Locations cannot be in different worlds.")
        val distance = start.distance(end)
        val v1 = start.toVector()
        val v2 = end.toVector()
        val vector = v2.clone().subtract(v1).normalize().multiply(2)
        var covered = 0
        while (covered < distance) {
            val particles = PacketPlayOutWorldParticles(EnumParticle.FLAME, true, v1.x.toFloat(), v1.y.toFloat(), v1.z.toFloat(), r.toFloat(), g.toFloat(), b.toFloat(), 255.toFloat(), 0, 10)
            for (player in Bukkit.getOnlinePlayers()) {
                (player as CraftPlayer).handle.playerConnection.sendPacket(particles)
            }
            covered += 2
            v1.add(vector)
        }
    }
}